src/basic_recipes/barplot.jl

```diff
@@ -17,23 +17,28 @@ Returns a Tuple of new y positions and offset arrays.
 - `ys`: The y-values passed to `barplot`.
 - `offset`: The `offset` parameter passed to `barplot`.
 """
-function bar_default_fillto(tf, ys, offset, in_y_direction)
-    return ys, offset
-end
-
-# `fillto` is related to `y-axis` transformation only, thus we expect `tf::Tuple`
-function bar_default_fillto(tf::Tuple, ys, offset, in_y_direction)
-    _logT = Union{typeof(log), typeof(log2), typeof(log10), Base.Fix1{typeof(log), <: Real}}
-    if in_y_direction && tf[2] isa _logT || (!in_y_direction && tf[1] isa _logT)
+function bar_default_fillto(tf, ys, offset, in_y_direction; log_clamp=false)
+    if log_clamp
         # x-scale log and !(in_y_direction) is equiavlent to y-scale log in_y_direction
         # use the minimal non-zero y divided by 2 as lower bound for log scale
-        smart_fillto = minimum(y -> y<=0 ? oftype(y, Inf) : y, ys) / 2
-        return clamp.(ys, smart_fillto, Inf), smart_fillto
+        _smart_log_clamp(ys)
     else
         return ys, offset
     end
 end
 
+"""
+    _smart_log_clamp(ys, ref=ys)::(ys_clamped, min_ref/2)
+
+Clamps the values of `ys` to a minimum value that is half the minimum non-zero value in `ref`.
```

Comment:
Is there a particular rationale for that value, or just arbitrary?

Reply:
This is heuristic from ROOT

Reply:
we could lower it to min_ref/5 or something, but I guess they felt /2 is more visually balanced in most cases.